homeassistant/components/media_player/sonos.py

```diff
@@ -63,9 +67,31 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
         _LOGGER.warning('No Sonos speakers found.')
         return False
 
-    add_devices(SonosDevice(hass, p) for p in players)
+    devices = [SonosDevice(hass, p) for p in players]
+    add_devices(devices)
     _LOGGER.info('Added %s Sonos speakers', len(players))
 
+    def group_players_service(service):
+        """Group media players, use player as coordinator."""
+        entity_id = service.data.get('entity_id')
+
+        if entity_id:
+            _devices = [device for device in devices
+                        if device.entity_name == entity_id]
```

Comment:
I'm pretty sure I've tried with that but it was giving me unexpected results during testing (it was concatenating some `_2` to the expected string), I looked into Entity too but I got lost so I went down this road... I'll have a look when I'm home and double check, sorry for the confusion